src/services/tx/tx-sender/sdk.ts

```diff
@@ -17,15 +22,44 @@ export const getAndValidateSafeSDK = (): Safe => {
   return safeSDK
 }
 
+export const switchWalletChain = async (onboard: OnboardAPI, chainId: string) => {
+  const chainIdInHex = hexValue(parseInt(chainId))
+
+  const wallet = getConnectedWallet(onboard.state.get().wallets)
+
+  if (wallet && isHardwareWallet(wallet)) {
+    await onboard.disconnectWallet({ label: wallet.label })
+    await connectWallet(onboard, { autoSelect: wallet.label })
```

Comment:
The user may connect to a wrong network here. We could recursively connect again if the newly connect wallet is on the wrong chain. What do you think?

Reply:
True, this is an issue with HW wallets specifically because it is possible to select any network when being triggered. Selecting a wrong network will result in a Core SDK error. I am not sure if it has to be done through recursion but we should definitely make sure that once we leave `switchWalletChain` that the wallet network really switched to the correct one.

Reply:
We could use the approach I proposed in #1729 as a fallback in that case. Basically allow to sign on the wrong chain – the only down downside is that typed signatures won't work, but HW wallets don't support them anyway, AFAIR.

Reply:
We were previously only throwing an error if the wallet disconnected. I've now added one for if the wallet connects to the wrong chain:

![wrong chain hw wallet](https://user-images.githubusercontent.com/20442784/225578341-17761480-e00e-4b65-8b68-9c1582cdec49.gif)

Reply:
Can't say this flow is the most intuitive. The wallet is already connected and you're suddenly popping up the connection again w/o any explanation.
It would confuse me.
Since an automatic switch cannot be performed for HW wallets, I would rather show a warning and a switch button instead re-connecting them silently.

Reply:
As per our discussion, we've added a warning regarding the reconnection attempt (only for hardware wallets):

![image](https://user-images.githubusercontent.com/20442784/225637575-cfb7ecdb-5f78-48b3-82da-6322ab7783a2.png)

Reply:
> As per our discussion, we've added a warning regarding the reconnection attempt (only for hardware wallets):
> 
> ![image](https://user-images.githubusercontent.com/20442784/225637575-cfb7ecdb-5f78-48b3-82da-6322ab7783a2.png)

is this a last version of implementation? I would propose a bit of design improvements for it :)